### PR TITLE
Masasa_Fixed the inactive and active issue sorting

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
# Description
Fixed the feature that enables the ability to filter by active, inactive and paused column
Screenshot 2024-02-28 at 8 45 45 AM
Screenshot 2024-02-28 at 8 45 54 AM

## Related PRS (if any):
This frontend PR is not realted to back end PR so Use development branch
…

## Main changes explained:
-  Introduced the allselected state and adjusted the filtering function

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73863365/66cc8570-2f9a-4f78-8d80-c8ccebf93d56

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ Other links→ user management→…
6. verify function that the filterighn works as shown in the video above

### CODE REVIEW

**Review by meetpadhiar4 — APPROVED** [review]
Hi, I tested your PR and it works correctly. I was able to filter the users based on their status and search users based on the input string. Please refer the video attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/ab6c7863-c99c-40b4-ae6b-928e8a8b8582

File: src/components/UserManagement/UserManagement.jsx [hunk 1]
```diff
@@ -235,34 +236,39 @@ class UserManagement extends React.PureComponent {
   };
 
   filteredUserList = userProfiles => {
-    let filteredList = userProfiles.filter(user => {
-      //Applying the search filters before creating each table data element
-      if (
-        (user.firstName.toLowerCase().indexOf(this.state.firstNameSearchText.toLowerCase()) > -1 &&
+    return userProfiles.filter(user => {
+      console.log('user', user);
```
Comment (FyisFe) [inline src/components/UserManagement/UserManagement.jsx:240]: Kindly remove `console.log()`

File: src/components/UserManagement/UserSearchPanel.jsx [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { boxStyle } from 'styles';
  * The search panel stateless component for user management grid
  */
 const UserSearchPanel = props => {
+  console.log('UserSearchPanel props', props);
```
Comment (FyisFe) [inline src/components/UserManagement/UserSearchPanel.jsx:8]: Kindly remove `console.log()`

File: src/components/UserManagement/UserManagement.jsx [hunk 3]
```diff
@@ -538,23 +544,36 @@ class UserManagement extends React.PureComponent {
...
         break;
       case 'inactive':
         active = false;
+        paused = false;
+        allSelected = false;
+        break;
+      case 'all':
+        allSelected = true;
+        active = false;
+        paused = false;
```
Comment (FyisFe) [inline src/components/UserManagement/UserManagement.jsx:564]: A neat picking, make all cases consistent

```js
active = false;
paused = false;
allSelected = true;
```

**Review by FyisFe — APPROVED** [review]
I tested it and all functionalities work well, please check the video below for details.

There are a few things could be improved, kindly resolve them before merging

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59299011/d07752ef-b240-49c1-aa8f-b1a53d0325e8

**Review by navyasrim — APPROVED** [review]
Hi, tested the PR and the search is working as mentioned, checked for all, active, inactive, paused and it worked as expected...Attached Image.
PR-2023

**Review by XiaohanMeng — APPROVED** [review]
Hi, I tested this PR and all functionalities are working as expected. The filter selection is fast, and the 'next page' pagination feature operates smoothly. The distinctions between active, inactive, and paused statuses are well-defined. Please refer the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/7507a4f4-3f0e-41eb-8cd8-151c81a19c0c

**Review by ShengjieMao — APPROVED** [review]
Filter worked as expected. And the codes looked great. While the instruction videos showed was the toggle tool rather than the filter function on my end. Not sure if the wrong  video was accidentally selected.  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/21ea7ef7-b255-44cc-b829-0edbbcd859d3

**Review by Gugor99 — APPROVED** [review]
### Report
Hi Masasa, I checked out your functionality and it seems to be working smoothly and as expected. As suggested, I logged in as admin and navigated to the user management's page. There I tried to filter the search result with each one of the three given options, and it was filtering the results as expected. Good job!
### Screenshots
Cattura1
Cattura2

**Comment by beblicarl** [comment]
This functionality works as intended and the code is great

[screencast-github.com-2024.03.01-18_05_41.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/aef03162-6728-4096-a4be-356747b0478f)

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**